_posts/2020-06-05-mirror-maker-2-eventhub.md

```diff
@@ -0,0 +1,290 @@
+    heartbeatConnector:
+      config:
+        heartbeats.topic.replication.factor: 1
+    checkpointConnector:
+      config:
+        checkpoints.topic.replication.factor: 1
+    topicsPattern: ".*"
+    groupsPattern: ".*"
+```
+
+Create the `KafkaMirrorMaker2` on the Kubernetes cluster saving the above resource into a `kafka-mirror-maker-2-from-eh.yaml` file.
+The Strimzi cluster operator takes care of it deploying Kafka Mirror Maker 2 using the above configuration.
+
+```shell
+kubectl apply -f kafka-mirror-maker-2-from-eh.yaml -n kafka
+```
+
+In this way, we can add another producer on the Event Hub side sending messages to the `testeh` event hub.
+We can use the `kafka-console-producer` for this purpose.
+
+```shell
+bin/kafka-console-producer.sh --bootstrap-server functiontesteh.servicebus.windows.net:9093 --topic testeh --producer.config kafka_eventhub.properties
+If you don't see a command prompt, try pressing enter.
+> "Hello from Strimzi Mirror Maker 2 sent to testeh Event Hub
+>
+```
+
+The producer is in addition to the one we used in the previous paragraph which is producing locally to `testeh` in the Kafka cluster.
+
+```shell
+kubectl -n kafka run kafka-producer -ti --image=strimzi/kafka:0.18.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-producer.sh --broker-list my-cluster-kafka-bootstrap:9092 --topic testeh
+If you don't see a command prompt, try pressing enter.
+>"Hello from Strimzi Mirror Maker 2 sent to testeh Kafka"
+>
+```
+
+We can also add a consumer on the local Kafka cluster getting messages on the corresponding mirrored `eventhub.testeh` topic.
+
+```shell
+kubectl -n kafka run kafka-consumer -ti --image=strimzi/kafka:0.18.0-kafka-2.5.0 --rm=true --restart=Never -- bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --whitelist .*testeh
+"Hello from Strimzi Mirror Maker 2 sent to testeh Event Hub"
+"Hello from Strimzi Mirror Maker 2 sent to testeh Kafka"
+```
+
+The consumer used in the previous paragrah will get both the messages as well.
+
+```shell
+bin/kafka-console-consumer.sh --bootstrap-server <eventhubs-namespace>.servicebus.windows.net:9093 --topic testeh --consumer.config kafka_eventhub.properties
+"Hello from Strimzi Mirror Maker 2 sent to testeh Event Hub"
+"Hello from Strimzi Mirror Maker 2 sent to testeh Kafka"
+```
+
+As you could have noticed, both the consumers have to subscribe to a topic pattern like `.*testeh` so that the one running locally can get messages from the local `testeh` and the mirrored `eventhub.testeh`; the other one connected to Azure Event Hub can get messages from the `testeh` event hub and the mirrored `my-cluster.testeh`.
+
+### Conclusion
+
+Kafka Mirror Makers 2 overcomes the limitations of Mirror Maker and it allows to do an active-active mirroring.
+Thanks to its integration in Strimzi, it's even simpler to setup this kind of use case with Azure Event Hub as well.
+
+Let us know what you are going to integrate!
```

Comment:
Perhaps add some links to how they might do this, whether it be the comment section of the blog, via twitter etc - might be useful to direct peoples interaction :)